pyproject.toml

```diff
@@ -52,7 +52,7 @@ pygsheets = "^2.0.4"
 PyYAML = "^5.4.1"
 rdflib = "^5.0.0"
 setuptools = "^52.0.0"
-synapseclient = "2.2.2"
+synapseclient = "^2.3.0"
```

Comment:
Not sure if we want to make this update just yet? The OAuth integration feature in the Data Curator App is still being [tested](https://github.com/Sage-Bionetworks/data_curator/tree/shiny-server-oauth) and we won't be ready to make this `synapseclient` transition until that's done. The reason is because version 2.3.0 deprecates the `sessionToken` parameter for the `login()` method which is currently the way in which the Data Curator App logs in.

Reply:
@sujaypatil96: I was going to downgrade back to version 2.2, but I realized that I actually need a feature that's been implemented in 2.3. Since session token support has been deprecated but not removed yet, I'll pin the version to `~2.3` ([source](https://python-poetry.org/docs/dependency-specification/)) to avoid updates to a version of `synapseclient` that removes support for session tokens.